ipa-core/src/utils/arraychunks.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{ff::Fp61BitPrime, secret_sharing::SharedValue};
+
+#[allow(dead_code)]
+pub trait ArrayChunkIterator: Iterator {
+    /// This function returns an iterator that yields arrays of size `L`.
+    /// When the amount of items in the iterator is not a multiple of `L`
+    /// the iterator fills the last array with zero elements.
+    fn chunk_array<const L: usize>(self) -> ArrayChunk<Self, L>
+    where
+        Self: Sized,
+    {
+        ArrayChunk { iter: self }
+    }
+}
+
+pub struct ArrayChunk<I, const L: usize> {
+    iter: I,
+}
+
+impl<I: Iterator<Item = Fp61BitPrime>> ArrayChunkIterator for I {}
+
+#[allow(clippy::while_let_on_iterator)]
+impl<I, const L: usize> Iterator for ArrayChunk<I, L>
+where
+    I: Iterator<Item = Fp61BitPrime>,
```

Comment:
Does this have to be `Item = Fp61BitPrime` or could it be `Item = T where T: Copy`?

Reply:
I was considering using a type that implements default or has a zero defined on it since I want to fill the rest of the array with some default or zero elements.

Reply:
Relying on an implementation of `Default` is probably OK, provided that we can do that for the types we care about.  A default of zero is expected, so no problem there.